tf2/mainpage.dox

```diff
@@ -1,36 +1,195 @@
 /**
 \mainpage
-\htmlinclude manifest.html
 
-\b tf2 is the second generation of the tf library.
+``tf2`` is the second generation of the tf library.
 
 This library implements the interface defined by tf2::BufferCore.
 
 There is also a Python wrapper with the same API that class this library using CPython bindings.
 
-\section codeapi Code API
+Some tutorials are available at <A HREF="http://docs.ros.org/en/rolling/Tutorials/Intermediate/Tf2/Tf2-Main.html">https://docs.ros.org/</A>.
+
+\section architecture Architecture
+
+``tf2`` is a transform library designed to provide implementation of interface that keeps track of multiple coordinate frames over time.
+``tf2`` maintains the relationship between coordinate frames in a tree structure buffered in time, and lets the user transform data between any two coordinate frames at any desired point in time.
+The high level goal is to allow developers and users not to have to worry about which coordinate frame any specific data is stored in.
+
+\subsection main-interface Main Interface
 
 The main interface is through the tf2::BufferCore interface.
 
-It uses the exceptions in exceptions.h and the Stamped datatype
-in transform_datatypes.h.
+It uses the exceptions in tf2/exceptions.h and the tf2::Stamped datatype in tf2/transform_datatypes.h.
 
-\section conversions Conversion Interface
+\subsection conversion-interface Conversion Interface
 
-tf2 offers a templated conversion interface for external libraries to specify conversions between
-tf2-specific data types and user-defined data types. Various templated functions in tf2_ros use the
-conversion interface to apply transformations from the tf server to these custom datatypes.
+tf2 offers a templated conversion interface for external libraries to specify conversions between ``tf2``-specific data types and user-defined data types.
+Various templated functions in ``tf2_ros`` use the conversion interface to apply transformations from the tf server to these custom datatypes.
 
 The conversion interface is defined in tf2/convert.h.
 
-Some packages that implement this interface:
+\subsection buffer-core-relations Buffer Core: Record and lookup relations between frames
+
+The ``tf2`` library implements the interface defined by tf2::BufferCore.
```

Comment:
```suggestion
The ``tf2`` library implements the interface defined by `tf2::BufferCore`.
```

Reply:
Fixed (along with the others) in 23e606d